### ISSUE TITLE
[Privacy] Per-User location sharing controls.

### ISSUE THREAD

**Issue by VixusFoxy** [Yellow-Dog-Man/Resonite-Issues#1121]
### Is your feature request related to a problem? Please describe.

Currently, the Resonite contacts list automatically shares user' locations when their session is public or under certain rules (like contacts, etc). This default setting poses concerns about user privacy and the lack of control over shared information.

### Describe the solution you'd like

I propose changing the default setting to an opt-in approach, allowing users to choose which information they share with their contacts. Specifically, I suggest implementing more granular privacy controls on a per-user basis, enabling users to customize the visibility of their location beyond the basic 'online status.' This solution aims to empower users and address privacy concerns even if there is always a technical workaround to this.

### Describe alternatives you've considered

**Status as 'Invisible' for all users:**
While this is a catch-all solution, it lacks granular control and may not be suitable for users with diverse contact lists. It stifles interaction because I will appear offline to everyone, while it could well be that I am attending an event that I don't necessarily want everyone to see directly from their contacts list, while also maintaining live communications in my contacts list. Showing as offline is not conducive to this.
    
**Enumerating public session info:**
While technically possible, this involves significant effort, the suggestion I present is not intended to be a 100% privacy control, it is however intended to let people choose what data **they** wish to volunteer, even if there are 'ways of figuring out anyway'.

**Discretion in adding contacts:**
While it could be argued that users should only add individuals they are comfortable sharing their location with, this does not account for the need to accommodate both personal and professional connections in a contact list.

**Using an alt account**
Certainly an option, but again, not user-friendly. Resonite's user base is diverse and multifaceted and everyone puts their own twist on how they interact with people. 

### Additional Context

Second Life has public worlds that can be enumerated for its users if one really wants. Yet, Linden Lab implemented controls in their contacts panel to allow users to decide on a per-case basis which information they share on their profile for users. 

Here are some examples of what that looks like in practice and these images serve to illustrate suggestion rather than being prescriptive of what such implementation should look like.

Granular controls in the user list
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/24366705/947e213f-5cf1-4521-8ddd-01f1df4df0e6)

Seeing people on the map (as an example of what sharing this would look like in SL's case)
_Second Life does not show locations in user profiles, but you can utilize the World Map to see where users who volunteer this information are at._ 
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/24366705/c1696e99-ebe6-47e0-b65e-14dba4dfd4c0)

**Comment by VixusFoxy** [comment]
TL;DR hypothetical: I don't want customers that I do commissions for **_to readily see_** that I am, for instance, attending religious services, support groups, or public workshop events. While of course they could happen into me at such an event, I just wish to be able to opt out from readily showing that in their contacts list.
